src/renderer/fonts/font_loader.rs

```diff
@@ -119,12 +119,9 @@ impl FontLoader {
     }
 
     fn load(&mut self, font_key: FontKey) -> Option<FontPair> {
-        let font_style = match (font_key.bold, font_key.italic) {
-            (true, true) => FontStyle::bold_italic(),
-            (false, true) => FontStyle::italic(),
-            (true, false) => FontStyle::bold(),
-            (false, false) => FontStyle::normal(),
-        };
+        let weight = font_key.weight.into();
+        let slant = font_key.slant.into();
+        let font_style = FontStyle::new(weight, Width::NORMAL, slant);
```

Comment:
> I think some parts of CachingShaper could be simplified by having a copy or reference to FontOptions, but if we do that, we should still provide a fn with custom FontKey, since it's used at some sections from CachingShaper::build_clusters() to retrieve FontPair with custom values to weight and slant, which ignores the ones at FontOptions, such as skia fallback and last resort fallback

FontKey can completely ignore existence of FontOptions. 

and this code will look like

```suggestion
        let font_style = FontStyle::new(self.options.get_weight(font_key.bold), Width::NORMAL, self.options.get_slant(font_key.italic));
```

get_weight and get_slant should be left in it's current state.

Skia Fallback is not a problem, it renders whatever you want (if you ask for oblique, it still renders something like italic).
Last Resort not a problem either, it has `bold: false, italic: false`, and it get's converted to expected normal weight and upright slant.

Reply:
Understood, I didn't know that